.github/workflows/quarkus.yml

```diff
@@ -0,0 +1,335 @@
+  MX_GIT_CACHE: refcache
+
+jobs:
+  build-quarkus-and-graalvm:
+    name: Nightly quarkus and GraalVM build
+    runs-on: ubuntu-18.04
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        fetch-depth: 1
+        path: graal
+    - uses: actions/checkout@v1
+      with:
+        repository: graalvm/mx.git
+        ref: master
+        path: mx
+    - uses: actions/checkout@v1
+      with:
+        repository: quarkusio/quarkus.git
+        ref: master
+        path: quarkus
+    # actions/setup-java setups a zulu community distibution of OpenJDK
+    # apt-get install openjdk-11-jdk installs the same zulu distribuion on github runners
+    # so we use joschi/setup-jdk instead with an sha for added security since tags can be updated
+    - uses: joschi/setup-jdk@9dc354d4a8f5781ba8880dc9e6a7bdcc7845e3e7
+      with:
+        java-version: 11
+    - uses: actions/cache@v1
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+    - uses: actions/cache@v1
+      with:
+        path: ~/.mx
+        key: ${{ runner.os }}-mx-${{ hashFiles('**/suite.py') }}
+        restore-keys: |
+          ${{ runner.os }}-mx-
+    - name: Get labsJDK11
+      run: |
+        java --version
+        wget --no-verbose "https://github.com/graalvm/labs-openjdk-11/releases/download/${JVMCI_VERSION}/labsjdk-ce-${JDK_VERSION}${JDK_VERSION_SUFFIX}-${JVMCI_VERSION}-linux-amd64.tar.gz"
+        tar xf labsjdk-ce-${JDK_VERSION}${JDK_VERSION_SUFFIX}-${JVMCI_VERSION}-linux-amd64.tar.gz
+        labsjdk-ce-${JDK_VERSION}-${JVMCI_VERSION}/bin/java --version
+    - name: Build graalvm native-image
+      run: |
+        export JAVA_HOME=${GITHUB_WORKSPACE}/labsjdk-ce-${JDK_VERSION}-${JVMCI_VERSION}
+        cd substratevm
+        ../../mx/mx build
+        mv $(../../mx/mx graalvm-home) ~/graaljdk
+        ~/graaljdk/bin/native-image --version
+    - name: Tar GraalVM
+      shell: bash
+      run: tar -czvf graaljdk.tgz -C ~ graaljdk
+    - name: Persist GraalVM build
+      uses: actions/upload-artifact@v1
+      with:
+        name: graaljdk
+        path: graaljdk.tgz
+    - name: Build quarkus
+      run: |
+        java --version
+        export JAVA_HOME=$(realpath $(dirname $(which java | xargs realpath))/../)
+        cd ../quarkus
+        eval ./mvnw -e -B -DskipTests -DskipDocs clean install
+    - name: Tar Quarkus
+      shell: bash
+      run: tar -czvf quarkus.tgz -C ../ quarkus
+    - name: Persist Quarkus
+      uses: actions/upload-artifact@v1
+      with:
+        name: quarkus
+        path: quarkus.tgz
+    - name: Tar Maven Repo
+      shell: bash
+      run: tar -czvf maven-repo.tgz -C ~ .m2/repository
+    - name: Persist Maven Repo
+      uses: actions/upload-artifact@v1
+      with:
+        name: maven-repo
+        path: maven-repo.tgz
+
+  native-tests:
+    name: Native Tests - ${{matrix.category}}
```

Comment:
worth noting that these matrix tests are a current snapshot from quarkus own native tests and they will change over time - we'll want to try and find a more stable way of defining the tests but for now this should do.

Reply:
+100

How would be achieve this though? Seems to me Quarkus should list/define these groups. However, don't know how Quarkus could maintain that list in a way that it doesn't break the moment a new integration test is added or removed.

Reply:
Started discussion on [the quarkus-dev mailing list](https://groups.google.com/forum/#!topic/quarkus-dev/_KZe1ydg4C0).

I suggest we proceed with merging this PR and when we find a better way of doing this we can create a new PR.